translation.rst

```diff
@@ -160,27 +157,121 @@ Now, if the language of the user's locale is French (e.g. ``fr_FR`` or ``fr_BE``
+    message, so your application can choose to disable the translator layer
+    and you will see a readable message.
+
+    Additionally, the ``php`` and ``yaml`` file formats support nested ids to
+    avoid repeating yourself if you use keywords instead of real text for your
+    ids:
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            symfony:
+                is:
+                    # id is symfony.is.great
+                    great: Symfony is great
+                    # id is symfony.is.amazing
+                    amazing: Symfony is amazing
+                has:
+                    # id is symfony.has.bundles
+                    bundles: Symfony has bundles
+            user:
+                # id is user.login
+                login: Login
+
+        .. code-block:: php
+
+            [
+                'symfony' => [
+                    'is' => [
+                        // id is symfony.is.great
+                        'great'   => 'Symfony is great',
+                        // id is symfony.is.amazing
+                        'amazing' => 'Symfony is amazing',
+                    ],
+                    'has' => [
+                        // id is symfony.has.bundles
+                        'bundles' => 'Symfony has bundles',
+                    ],
+                ],
+                'user' => [
+                    // id is user.login
+                    'login' => 'Login',
+                ],
+            ];
+
 The Translation Process
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 To actually translate the message, Symfony uses a simple process:
 
-* The ``locale`` of the current user, which is stored on the request is determined;
+#. The ``locale`` of the current user, which is stored on the request is determined;
 
-* A catalog (e.g. big collection) of translated messages is loaded from translation
-  resources defined for the ``locale`` (e.g. ``fr_FR``). Messages from the
-  :ref:`fallback locale <translation-fallback>` are also loaded and
-  added to the catalog if they don't already exist. The end result is a large
-  "dictionary" of translations.
+#. A catalog (i.e. big collection) of translated messages is loaded from translation
+   resources defined for the ``locale`` (e.g. ``fr_FR``). Messages from the
+   :ref:`fallback locale <translation-fallback>` are also loaded and
+   added to the catalog if they don't already exist. The end result is a large
+   "dictionary" of translations.
 
-* If the message is located in the catalog, the translation is returned. If
-  not, the translator returns the original message.
+#. If the message is located in the catalog, the translation is returned. If
+   not, the translator returns the original message.
 
 When using the ``trans()`` method, Symfony looks for the exact string inside
 the appropriate message catalog and returns it (if it exists).
 
+.. _using-message-domains:
+
+Using Message Domains
```

Comment:
I think we should recommend to NOT create domains unless your app needs are special or too complex. Domains complicate everything:

* It makes translation much more verbose because you always need to pick the domain when translating any content
* You need to update all your templates too to choose the domain
* It increases the "mental complexity" or your app: whenever you translate something, you need to remember (or look for) in which domain is that content defined.

Some developers wrongly think that domains will help them "decouple" translations into "small files" and make things more manageable and faster. But creating just 1 `messages` domain is the clever thing to do in most apps according to my experience.

Reply:
Is there a good reason to use multiple domains in your application? I can see how it can be usefull for third party bundles, but I fail providing a "but if you're in case ..., you might want to do this!" for application developers. And I don't like documenting something that you discourage to use.

Maybe with large translation teams? E.g. if you want to split things by specific "domain" (bounded contexts), as domain experts provide the translations for their specific domain?